core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/cluster/StopStartUpscaleGetStartedInstancesHandler.java

```diff
@@ -0,0 +1,130 @@
+import com.sequenceiq.cloudbreak.cloud.event.instance.StopStartUpscaleGetStartedInstancesResult;
+import com.sequenceiq.cloudbreak.cloud.handler.CloudPlatformEventHandler;
+import com.sequenceiq.cloudbreak.cloud.init.CloudPlatformConnectors;
+import com.sequenceiq.cloudbreak.cloud.model.CloudInstance;
+import com.sequenceiq.cloudbreak.cloud.model.CloudVmInstanceStatus;
+import com.sequenceiq.cloudbreak.cloud.model.InstanceStatus;
+import com.sequenceiq.cloudbreak.cluster.api.ClusterStatusService;
+import com.sequenceiq.cloudbreak.cluster.status.ExtendedHostStatuses;
+import com.sequenceiq.cloudbreak.converter.CloudInstanceIdToInstanceMetaDataConverter;
+import com.sequenceiq.cloudbreak.converter.spi.InstanceMetaDataToCloudInstanceConverter;
+import com.sequenceiq.cloudbreak.domain.stack.Stack;
+import com.sequenceiq.cloudbreak.domain.stack.instance.InstanceMetaData;
+import com.sequenceiq.cloudbreak.service.cluster.ClusterApiConnectors;
+import com.sequenceiq.cloudbreak.service.stack.RuntimeVersionService;
+import com.sequenceiq.cloudbreak.service.stack.StackService;
+
+import reactor.bus.Event;
+import reactor.bus.EventBus;
+
+@Component
+public class StopStartUpscaleGetStartedInstancesHandler implements CloudPlatformEventHandler<StopStartUpscaleGetStartedInstancesRequest> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(StopStartUpscaleGetStartedInstancesHandler.class);
+
+    @Inject
+    private CloudPlatformConnectors cloudPlatformConnectors;
+
+    @Inject
+    private ClusterApiConnectors clusterApiConnectors;
+
+    @Inject
+    private StackService stackService;
+
+    @Inject
+    private RuntimeVersionService runtimeVersionService;
+
+    @Inject
+    private CloudInstanceIdToInstanceMetaDataConverter cloudInstanceIdToInstanceMetaDataConverter;
+
+    @Inject
+    private InstanceMetaDataToCloudInstanceConverter instanceMetaDataToCloudInstanceConverter;
+
+    @Inject
+    private EventBus eventBus;
+
+    @Override
+    public Class<StopStartUpscaleGetStartedInstancesRequest> type() {
+        return StopStartUpscaleGetStartedInstancesRequest.class;
+    }
+
+    @Override
+    public void accept(Event<StopStartUpscaleGetStartedInstancesRequest> event) {
+        StopStartUpscaleGetStartedInstancesRequest request = event.getData();
+        LOGGER.info("StopStartUpscaleGetStartedInstancesHandler: {}", event.getData().getResourceId());
+
+        CloudContext cloudContext = request.getCloudContext();
+
+        try {
+            CloudConnector<?> connector = cloudPlatformConnectors.get(cloudContext.getPlatformVariant());
+            AuthenticatedContext ac = getAuthenticatedContext(request, cloudContext, connector);
+
+            Stack stack = stackService.getByIdWithLists(request.getResourceId());
+
+            List<InstanceMetaData> startedInstances = cloudInstanceIdToInstanceMetaDataConverter.getNotDeletedAndNotZombieInstances(
```

Comment:
This look up of instances known to CB already happens in the Action. Can likely be removed from StopStartUpscaleActions that triggers this handled.